File: src/ikj/main/ioke/lang/parser/Level.java
/*
 * See LICENSE file in distribution for copyright and licensing information.
 */
package ioke.lang.parser;

import ioke.lang.IokeObject;

/**
 * @author <a href="mailto:ola.bini@gmail.com">Ola Bini</a>
 */
final class Level {
    public static enum Type {
        REGULAR, UNARY, ASSIGNMENT, INVERTED
    }

    final int precedence;
    final IokeObject operatorMessage;
    final Level parent;
    final Type type;

    Level(int precedence, IokeObject op, Level parent, Type type) {
        this.precedence = precedence;
        this.operatorMessage = op;
        this.parent = parent;
        this.type = type;
    }

    public String toString() {
        return "Level<" + precedence + ", " + operatorMessage + ", " + type + ", " + parent + ">";
    }
}// Level

File: src/ikc/main/Ioke.Lang/Parser/Level.cs

namespace Ioke.Lang.Parser {
    using Ioke.Lang;

    public class Level {
        public enum Type {REGULAR, UNARY, ASSIGNMENT, INVERTED};

        readonly internal int precedence;
        readonly internal IokeObject operatorMessage;
        readonly internal Level parent;
        readonly internal Type type;

        internal Level(int precedence, IokeObject op, Level parent, Type type) {
            this.precedence = precedence;
            this.operatorMessage = op;
            this.parent = parent;
            this.type = type;
        }

        public override string ToString() {
            return "Level<" + precedence + ", " + operatorMessage + ", " + type + ", " + parent + ">";
        }
    }
}
